dbms/src/Core/SettingsCommon.h

```diff
@@ -34,6 +34,7 @@ struct SettingNumber
 {
     Type value;
     bool changed = false;
+    bool immutable = false;
```

Comment:
I think functions `SettingNumber::isImmutable()` and `SettingNumber::setImmutable()` will look good here because `immutability` is some kind of state of `SettingNumber` and it's usually right to provide a class with functions to change its own state.